### DISCUSSION TITLE
Live/Post Contest logging?

### DISCUSSION THREAD

**Discussion by Werzi2001** [magicbug/Cloudlog#1441] [Ideas / Feature Requests]
I'm planning to take part in the WPX contest this weekend. As it happens my current situation only allows to go QRV outdoors and the internet connection is not that good. So basically I'll do my log on paper and add it later to Cloudlog. Now I noticed that there is not Post Contest Logging at the moment (only Live).

Is it just me or would that be a useful addition? I could implement it, just wanted to ask if it is needed. I would add it in the same way as it was done for "Live QSO" and "Post QSO".

Furthermore I was wondering if it would make sense to be able to switch between live/post on-the-fly. I was thinking about adding a small button next to date/time which just switches between post/live. Maybe that would be useful also for "Live QSO" and "Post QSO". So basically not having these two menus but just a button to switch on-the-fly in the form.

What do you think?

**Comment by AndreasK79** [comment]
@Werzi2001 that sounds like a good plan to me. You get my vote for implementing it in contest logging as you described.
As for changing the normal way, I guess it's up to @magicbug, but I would find this as a nice simplification.

**Comment by magicbug** [comment]
Not sure I ever do "post contest" logging but maybe that's just me :) however, if a switch can be implemented so that it looks nice within the interface it might work.

**Reply by DF4OR** [reply]
+1 for a post contest logging option. Occasionally I go portable, no internet, or not even something like a computer. Logging on paper, and would like to enter these qsos post-event.

Thanks & 73,
Ekki, DF4OR

**Reply by AndreasK79** [reply]
This has been added to the station_logbooks branch.

**Comment by Cadair** [comment]
I ended up doing the 144MHz Backpackers contest this weekend, and logging the old fashioned way. I am now struggling to digitise my log because of the lack of contest fields in the Post QSO form and the live clock in the Contest Logging.

I might have a quick poke at fixing this, I assume it's a case of finding the right bit of javascript to disable?

**Reply by Cadair** [reply]
OK, so I just nastily hacked this so I can enter my contest log by stripping the classes used by [the problematic js functions](https://github.com/magicbug/Cloudlog/blob/master/assets/js/sections/contesting.js#L54-L68).

If I wanted to implement a toggle for these functions, is the best way to approach it to make the functions check the toggle or have the toggle remove the classes from the element the functions use? (Javascript, but especially frontend js is not my bag!)

**Reply by AndreasK79** [reply]
@Cadair post contest logging was merged into station logbooks (v2 branch). @Werzi2001 tried to implement a toggle, but found it was easier just to implement post / live contest logging.
